backend/core/src/main/java/org/sonarsource/sonarlint/core/analysis/AnalysisExtraPropertiesService.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonarsource.sonarlint.core.analysis;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.initialize.InitializeParams;
+
+@Named
+@Singleton
+public class AnalysisExtraPropertiesService {
```

Comment:
I wonder if we need a separate service for that, it's related to the analysis so reusing the existing `AnalysisService` might be sufficient? In fact you could transform this class into a repository and use it from another service

Reply:
Refactored as discussed